amplipi/streams.py

```diff
@@ -848,56 +848,92 @@ def connect(self, src):
+                '--gstout-audiodevice', utils.real_output_device(src), '--gstout-initial-volume-db',
+                '0.0', '-p', f'{portnum}', '-u', f'{self._uuid}',
+                '-f', f'{self.name}']
+    meta_args = [ sys.executable,
+                 f'{utils.get_folder("streams")}/dlna_meta.py',
+                 f'{self.name}',
+                 f'{self._src_config_folder}/cmd',
+                 f'{self._src_config_folder}/meta.json',
+                 self._src_web_folder,
+                ]
+                # '-d']
+
+    self.proc = subprocess.Popen(args=dlna_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    self._metadata_proc = subprocess.Popen(args=meta_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    print(f'{self.name} connected to {src}')
     self._connect(src)
 
   def disconnect(self):
     if self._is_running():
-      os.killpg(os.getpgid(self.proc.pid), signal.SIGKILL)
-      self.proc2.kill()
+      utils.careful_proc_shutdown(self._metadata_proc, "dlna metadata process")
+      utils.careful_proc_shutdown(self.proc, "dlna process")
+      if self._fifo_open:
+        os.close(self._fifo)
     self._disconnect()
-    self.proc = None
-    self.proc2 = None
+    self._metadata_proc = None
+    self.dlna_proc = None
 
   def info(self) -> models.SourceInfo:
-    src_config_folder = f'{utils.get_folder("config")}/srcs/{self.src}'
-    loc = f'{src_config_folder}/currentSong'
     source = models.SourceInfo(name=self.full_name(), state=self.state, img_url='static/imgs/dlna.png')
     try:
-      with open(loc, 'r', encoding='utf-8') as file:
-        for line in file.readlines():
-          line = line.strip()
-          if line:
-            d = ast.literal_eval(line)
-        if 'state' in d:
-          source.state = d['state']
-        if 'album' in d:
-          source.album = d['album']
-        if 'artist' in d:
-          source.artist = d['artist']
-        if 'title' in d:
-          source.track = d['title']
-        return source
-    except Exception:
+
+      data = json.load(open(f'{self._src_config_folder}/meta.json'))
+      print(data)
+      source.state = data.get('state', 'stopped') if data else 'stopped'
+      if source.state != 'stopped': # if the state is stopped, just use default values
+        source.artist = data.get('artist', '')
+        source.track = data.get('title', '')
+        source.album = data.get('album', '')
+        if data.get('album_art', '') != '':
+          source.img_url = f'generated/{self.src}/{data.get("album_art", "")}'
+
+      source.supported_cmds=self.supported_cmds # set supported commands only if we hear back from the DLNA server
+      self._got_data = True
+    except Exception as e:
+      if self._got_data: # ignore if we havent gotten data yet since we're still waiting for the metadata process to start
+        print(f'Error getting DLNA info: {e}')
```

Comment:
I wonder what will land first - this or #661 :thinking:

Reply:
I won the race
Let me know if you need help implementing anything to have this branch be in line with #661, though it's fairly straightforward